plugin/pkg/scheduler/schedulercache/interface.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulercache
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// Cache collects pods' information and provides node-level aggregated information.
+// It's intended for generic scheduler to do efficient lookup.
+// Cache's operations are pod centric. It incrementally updates itself based on pod event.
+// Pod events are sent via network. We don't have guaranteed delivery of all events.
```

Comment:
"We don't have guaranteed delivery of all events".

What do you mean by that?
In fact if you have a watch, you are guaranteed that all events will be delivered. The only potential problem is that we may be slow (or loose internet connection) and may be too old in history. In such case, what we should do is to relist all pods, to get the current state from etcd.
Is that what you meant by that? If so can you please clarify?